frontend/src/lib/integrations/slackIntegrationLogic.ts

```diff
@@ -7,6 +7,33 @@ import { SlackChannelType } from '~/types'
 
 import type { slackIntegrationLogicType } from './slackIntegrationLogicType'
 
+export class SlackChannels {
```

Comment:
This looks tricky to my eye to decern and is really different to our frontend pattern. Could this not just be a function that returns an object?

Reply:
I don't know how to make this better in Kea. I wanted to hide the two "internal" lists of channels to make sure they are not used by accident. I implemented it as a function in the beginning, but that meant I had the internal lists exposed in the reducer state. Any suggestions on how I can get around it?

I prefer having the merge channels logic wrapped in a small class that has one obvious to use interface rather than leaving the internal state open.